src/Build/Collections/RetrievableEntryHashSet/HashSet.cs

```diff
@@ -1290,338 +1288,6 @@ internal bool EntriesAreReferenceEquals(RetrievableEntryHashSet<T> other)
             return true;
         }
 
-#if NEVER
```

Comment:
I am really unsure why this was in there, I tracked it back to the initial commit to msbuild from vs2015. It looks like its 'NEVER' called?

Reply:
This sort of thing makes me nervous. I'll try to find my notes on how to see history from before vs2015.

Reply:
Good news and bad news. I found the commit. It seems to have come in September 2012, and I have a commit hash. Unfortunately, the entire message was basically "Dev11 RTM" plus a link to where they got it. I can find the file, but it's outside git, and I don't know how to get the blame equivalent for that.

Reply:
That is back to vs2012 Release to Manufacturing. Times flies. 

Does `#if NEVER` actually never get compiled? If not, then this code won't ever be run. MOST of it is private. with the "equals" being internal. As far as I could tell it wasn't referenced anywhere.

Reply:
If I saw this, I would normally feel pretty confident that we can delete it all, but weird things happen in MSBuild sometimes, and it's theoretically possible that NEVER is a variable that means "only in my special debug mode" or something like that. This just feels too obviously like dead code 😉

Some people are coming back next week, so I should (hopefully) be able to get an answer then. I'll also hopefully be able to start merging things then...lots of PRs out that I think are ready to go in.

Reply:
I reverted the change and now the 'unused' constant is wrapped in an `#if NEVER`. Logically grouping them.

Reply:
From @ladipro:
Looks like it was used for commenting out debug-only or old code. I'm all for deleting these, unless they were added recently and someone is planning to resurrect the code which I don't think is the case.

I think we're good for deleting it (🥳)

Reply:
Actually, hold on for a bit.

Reply:
Sorry I got distracted—answer seems to be that we should avoid touching that file—and that was the point of the #if NEVER blocks. They aren't used, but if we need to merge in more changes from the original version of that file at some point, we want it as similar as possible without adding extra gunk to MSBuild assemblies. Adding #if NEVER means nothing unnecessary is added but also doesn't change the file very much. I'd suggest avoiding any changes to the file at all, sadly.